### PR TITLE
MCP-1918:  Controller shell for MAS Request 

### PR DESCRIPTION
**Pull request by dimitri-amida**
# MAS Controller

### What was the problem?

"The VA Mailroom gets a claim from a veteran, it is digitized and a notification is sent to VRO if the claim is eligible for automation.
It would take more than an hour for MAS to collect the health history related to the claim from various data sources (CAPRI, LHAPI etc). This API facilitates MAS to send the notification to VRO. 

VRO will wait for an hour and check if the health evidence is collected by the MAS by calling an API. On a positive response from the call, VRO will make another api call to MAS to collect the health evidence. 

VRO will validate the health history for an evidence related to the claim

VRO will compile the evidence in the standard PDF format and uploads it into the e-folder

VRO will call an API to order exam if the evidence is not sufficient for the claim

VRO will update the status if the claim in the VBMS by calling an API

### How does this fix it?

Controller to receive initial MAS request

### Jira Tickets

- [MCP-1918](https://amida.atlassian.net/browse/MCP-1918)

## How to test this PR

- Call API  /v1/notifyVROAutomatedClaimDetails
- If the request is invalid you will receive a 500 response, otherwise 200
- NOTE: There is not business logic performed by the controller at this stage

## Reminders

- Review [Pull-Requests](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests) guidelines
- [ ] If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) and
[Roadmap](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Roadmap) wiki pages
- [ ] If changing software behavior, post a link to the PR in Slack channel #benefits-virtual-regional-office

### CODE REVIEW

**Comment by dimitri-amida** [comment]
need to move branch

**Review by rajesh-Krishnasamy — APPROVED** [review]

File: api/src/main/java/gov/va/vro/api/model/mas/ClaimDetailConditions.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ClaimDetailConditions {
+
+  private String name;
+
+  @NotBlank(message = "Diagnostic Code is required")
+  private String diagnosticcode;
+
+  private String disabilityactiontype;
```
Comment (au2) [inline api/src/main/java/gov/va/vro/api/model/mas/ClaimDetailConditions.java:21]: Why do we have these all lower case? Isn't Java convention lower camel case? If MAS specified as such I rather use @JsonNaming(PropertyNamingStrategies.LowerCaseStrategy).

File: api/src/main/java/gov/va/vro/api/requests/MASClaimDetailsRequest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import lombok.Getter;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
+@Builder
+@Getter
+@Schema(name = "MASClaimDetailsRequest", description = "Initiate a MAS request")
+public class MASClaimDetailsRequest {
```
Comment (au2) [inline api/src/main/java/gov/va/vro/api/requests/MASClaimDetailsRequest.java:16]: I would use MAS to get rid of build warnings.

File: api/src/main/java/gov/va/vro/api/resources/MasResource.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            responseCode = "500",
+            description = "Data Access Server Error",
+            content = @Content(schema = @Schema(hidden = true))),
+        @ApiResponse(
+            responseCode = "404",
+            description = "No evidence found",
+            content = @Content(mediaType = "application/json"))
+      })
+  @Timed(value = "mas-claim-request")
+  @Tag(name = "MAS Claim Request")
```
Comment (au2) [inline api/src/main/java/gov/va/vro/api/resources/MasResource.java:48]: You need to add this to the global Tag. See OpenApiConfiguration.java in app/config. This is also a build warning.

**Review by au2 — CHANGES_REQUESTED** [review]
Please fix all build warnings.

File: api/src/main/java/gov/va/vro/api/requests/MASClaimDetailsRequest.java (hunk 2 above)
Comment (dimitri-amida) [reply api/src/main/java/gov/va/vro/api/requests/MASClaimDetailsRequest.java]: done

File: api/src/main/java/gov/va/vro/api/resources/MasResource.java (hunk 3 above)
Comment (dimitri-amida) [reply api/src/main/java/gov/va/vro/api/resources/MasResource.java:48]: done

File: api/src/main/java/gov/va/vro/api/model/mas/ClaimDetailConditions.java (hunk 1 above)
Comment (dimitri-amida) [reply api/src/main/java/gov/va/vro/api/model/mas/ClaimDetailConditions.java:21]: Setting the naming strategy on the file does not work. I have to specify the property for each field

**Review by au2 — CHANGES_REQUESTED** [review]
Can you fix these warnings too? Just a naming warnings but I do not want to have them during build. We just fixed bunch of them.
[ant:checkstyle] [WARN] /private/work/va/abd-vro/api/src/main/java/gov/va/vro/api/resources/MasResource.java:49:36: Abbreviation in name 'notifyVROAutomatedClaimDetails' must contain no more than '1' consecutive capital letters. [AbbreviationAsWordInName]
[ant:checkstyle] [WARN] /private/work/va/abd-vro/api/src/main/java/gov/va/vro/api/requests/MASClaimDetailsRequest.java:16:14: Abbreviation in name 'MASClaimDetailsRequest' must contain no more than '1' consecutive capital letters. [AbbreviationAsWordInName]
Checkstyle rule violations were found. See the report at: file:///private/work/va/abd-vro/api/build/reports/checkstyle/main.html
Checkstyle files with violations: 2
Checkstyle violations by severity: [warning:2]

> Task :app:checkstyleTest
[ant:checkstyle] [WARN] /private/work/va/abd-vro/app/src/test/java/gov/va/vro/MasControllerTest.java:20:8: Abbreviation in name 'notifyVROAutomatedClaimDetails_invalidRequest' must contain no more than '1' consecutive capital letters. [AbbreviationAsWordInName]
[ant:checkstyle] [WARN] /private/work/va/abd-vro/app/src/test/java/gov/va/vro/MasControllerTest.java:30:8: Abbreviation in name 'notifyVROAutomatedClaimDetails_validRequest' must contain no more than '1' consecutive capital letters. [AbbreviationAsWordInName]
Checkstyle rule violations were found. See the report at: file:///private/work/va/abd-vro/app/build/reports/checkstyle/test.html
Checkstyle files with violations: 1
Checkstyle violations by severity: [warning:2]

**Comment by dimitri-amida** [comment]
@au2 if we want checkstyle violations to be enforced, we should be failing the build on them. Otherwise, it's very easy to miss these warnings.

**Review by au2 — APPROVED** [review]

**Pull request merged**